src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/FullAdmissionResponseContentProvider.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.hmcts.cmc.claimstore.documents.content;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.domain.models.PaymentOption;
+import uk.gov.hmcts.cmc.domain.models.RepaymentPlan;
+import uk.gov.hmcts.cmc.domain.models.response.FullAdmissionResponse;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.cmc.claimstore.services.staff.content.RepaymentPlanContentProvider.create;
+
+@Component
+public class FullAdmissionResponseContentProvider {
+
+    private final StatementOfMeansContentProvider statementOfMeansContentProvider;
+
+    public FullAdmissionResponseContentProvider(
+        StatementOfMeansContentProvider statementOfMeansContentProvider
+    ) {
+        this.statementOfMeansContentProvider = statementOfMeansContentProvider;
+    }
+
+    public Map<String, Object> createContent(FullAdmissionResponse fullAdmissionResponse) {
+        requireNonNull(fullAdmissionResponse);
+
+        RepaymentPlan repaymentPlan = null;
+        PaymentOption type = fullAdmissionResponse.getPaymentOption();
+        Optional<RepaymentPlan> optionalRepaymentPlan = fullAdmissionResponse.getRepaymentPlan();
+        if (optionalRepaymentPlan.isPresent()) {
+            repaymentPlan = optionalRepaymentPlan.get();
+        }
+        ImmutableMap.Builder<String, Object> contentBuilder = ImmutableMap.builder();
+        contentBuilder.put("responseTypeSelected", fullAdmissionResponse.getResponseType().getDescription());
+
+        switch (type) {
+            case IMMEDIATELY:
+                contentBuilder.put("paymentOption", type.getDescription());
+                break;
+            case FULL_BY_SPECIFIED_DATE:
+                contentBuilder.put("paymentOption", type.getDescription());
+                fullAdmissionResponse.getStatementOfMeans().ifPresent(
```

Comment:
Content creation for SoM can be done outside the switch statement as well on the assumption that SoM will be present in response only when it should be there.

Reply:
cleaned.